Mathlib/Data/List/Sort.lean

```diff
@@ -191,6 +190,86 @@ alias ⟨_, _root_.Monotone.ofFn_sorted⟩ := sorted_le_ofFn_iff
 
 end Monotone
 
+end List
+
+open List
+
+namespace StrictMono
+
+variable {α β : Type*} [LinearOrder α] [Preorder β] {f : α → β} {l : List α}
+
+theorem sorted_le_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· ≤ ·) ↔ l.Sorted (· ≤ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_ge_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· ≥ ·) ↔ l.Sorted (· ≥ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_lt_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· < ·) ↔ l.Sorted (· < ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
+
+theorem sorted_gt_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· > ·) ↔ l.Sorted (· > ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
```

Comment:
What do you think of marking these `simp`, which would remove the need for the other special cases below?

Reply:
See above. Note for the future: I'm not sure if `StrictMono` is the right notion for partial orders. Should we change it to "le iff le" with a constructor from "lt implies lt" in case of a linear order?

Reply:
cc @YaelDillies